src/DebugPoints/DebugPointManager.class.st

```diff
@@ -0,0 +1,374 @@
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #read
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableWriteNamed: aSlotNameSymbol [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #write
+		  withBehaviors: {  }
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass inClass: aClass onVariableWriteNamed: aSlotNameSymbol withBehaviors: aListOfBehaviorClasses [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  inClass: aClass
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #write
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'announcements' }
+DebugPointManager class >> notifyDebugPointAdded: aDebugPoint [
+
+	| announcement |
+	announcement := DebugPointAdded 
+		on: aDebugPoint
+		nodes: aDebugPoint link nodes.
+	SystemAnnouncer uniqueInstance announce: announcement
+]
+
+{ #category : 'announcements' }
+DebugPointManager class >> notifyDebugPointChanged: aDebugPoint [
+	
+	| announcement |
+	announcement := DebugPointChanged
+		on: aDebugPoint
+		nodes: aDebugPoint link nodes.
+	SystemAnnouncer uniqueInstance announce: announcement
+]
+
+{ #category : 'announcements' }
+DebugPointManager class >> notifyDebugPointRemoved: aDebugPoint fromNodes: nodes [
+	"nodes have to be seperate because they will be removed from the debugpoint before this method is called"
+	| announcement |
+	announcement := DebugPointRemoved
+		on: aDebugPoint 
+		nodes: nodes.
+	SystemAnnouncer uniqueInstance announce: announcement.
+
+		
+
+]
+
+{ #category : 'announcements - registration' }
+DebugPointManager class >> registerInterestToSystemAnnouncement [
+	<systemEventRegistration>
+
+	SystemAnnouncer uniqueInstance unsubscribe: self.
+	SystemAnnouncer uniqueInstance weak when: MethodRemoved send: #handleMethodRemoved: to: self.
+	SystemAnnouncer uniqueInstance weak when: MethodModified send: #handleMethodModified: to: self.
+	SystemAnnouncer uniqueInstance weak when: ClassRemoved send: #handleClassRemoved: to: self
+]
+
+{ #category : 'removing' }
+DebugPointManager class >> removeFromMethod: aMethod [
+
+	DebugPoint all copy do: [ :debugPoint | 
+		debugPoint link methods
+			detect: [ :m | m == aMethod ]
+			ifFound: [ debugPoint removeFromMethod: aMethod ] ]
+]
```

Comment:
Same remark, for future work: the debug point model should probably know the methods it affects without having to ask them to the metalink.